python/ray/tune/tune.py

```diff
@@ -303,6 +303,16 @@ def run(
             "removing this argument from your call to `tune.run()`"
         )
 
+    # Starting deprecation in ray 1.10.
+    if os.environ.get("TUNE_TRIAL_RESULT_WAIT_TIME_S") is not None:
+        raise DeprecationWarning("`TUNE_TRIAL_RESULT_WAIT_TIME_S` is deprecated.")
+
+    if os.environ.get("TUNE_TRIAL_STARTUP_GRACE_PERIOD") is not None:
+        raise DeprecationWarning("`TUNE_TRIAL_STARTUP_GRACE_PERIOD` is deprecated.")
+
+    if os.environ.get("TUNE_TRIAL_RESULT_WAIT_TIME_S") is not None:
+        raise DeprecationWarning("`TUNE_TRIAL_RESULT_WAIT_TIME_S` is deprecated.")
+
```

Comment:
```suggestion
    # Starting deprecation in ray 1.10.
    if os.environ.get("TUNE_TRIAL_RESULT_WAIT_TIME_S") is not None:
        raise DeprecationWarning("`TUNE_TRIAL_RESULT_WAIT_TIME_S` is deprecated.")

    if os.environ.get("TUNE_TRIAL_STARTUP_GRACE_PERIOD") is not None:
        raise DeprecationWarning("`TUNE_TRIAL_STARTUP_GRACE_PERIOD` is deprecated.")

    if os.environ.get("TUNE_TRIAL_RESULT_WAIT_TIME_S") is not None:
        raise DeprecationWarning("`TUNE_TRIAL_RESULT_WAIT_TIME_S` is deprecated.")

```

Maybe we can use warnings.warn instead? I don't think raising an exception here is appropriate.

Reply:
Changed to `warnings.warn`